crates/frontend/src/api.rs

```diff
@@ -276,3 +278,21 @@ pub async fn fetch_types(
         .await
         .map_err(err_handler)
 }
+
+pub async fn fetch_workspaces(
+    filters: &PaginationParams,
+) -> Result<PaginatedResponse<Workspace>, ServerFnError> {
+    let client = reqwest::Client::new();
+    let host = use_host_server();
+    let url = format!("{}/workspaces?{}", host, filters.to_string());
+    let response: PaginatedResponse<Workspace> = client
+        .get(url)
+        .header("x-tenant", "dev")
```

Comment:
@mahatoankitkumar you can remove this and add /workspace to tenant-middleware-exclusion-list env. nab